File: android/lib/interfaces/observer-event-description.dart
import 'package:polydiff/interfaces/event-description.dart';
import 'package:polydiff/pages/replay-page.dart';

class ObserverEventDescription extends EventDescription{
  
  final Rectangle rectangle;
  final String color;

  const ObserverEventDescription(super.time, this.rectangle, this.color);

  ObserverEventDescription.fromJson(Map<String, dynamic> json)
      : rectangle = json['rectangle'],
        color = json['color'],
        super.fromJson(json);

  Map<String, dynamic> toJson() {
    return {
      'time': time,
      'rectangle': rectangle,
      'color': color,
    };
  }

  @override
  play(ReplayPage replayPage) {
    print("ObserverEventDescription.play() called.");
  }
  
}

class Rectangle{
  int startX;
  int startY;
  int width;
  int height;
  Rectangle(this.startX, this.startY, this.width, this.height);
}

File: client/src/app/classes/observer-event-description.ts
import { ReplayPageComponent } from "@app/pages/replay-page/replay-page.component";
import { EventDescription } from "./event-description";

export class ObserverEventDescription extends EventDescription {
    private rectangle: {startX:number, startY:number, width:number, height:number};
    private color: string;
    constructor(time: number,rectangle:{startX:number, startY:number, width:number, height:number}, color: string) {
        super(time);
        this.rectangle = rectangle;
        this.color = color;

    }
    play(replayPage: ReplayPageComponent) {
        replayPage.gamePageElem.canvas3.drawReceivedRectangle(this.rectangle, this.color);
        replayPage.gamePageElem.canvas4.drawReceivedRectangle(this.rectangle, this.color);
    }
}